lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/AwsSdkV2HttpClientStats.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.s3;
+
+import com.google.errorprone.annotations.ThreadSafe;
+import io.airlift.stats.DistributionStat;
+import io.airlift.stats.TimeStat;
+import org.weakref.jmx.Managed;
+import org.weakref.jmx.Nested;
+import software.amazon.awssdk.metrics.SdkMetric;
+
+import java.time.Duration;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static software.amazon.awssdk.http.HttpMetric.AVAILABLE_CONCURRENCY;
+import static software.amazon.awssdk.http.HttpMetric.LEASED_CONCURRENCY;
+import static software.amazon.awssdk.http.HttpMetric.PENDING_CONCURRENCY_ACQUIRES;
+
+@ThreadSafe
+public class AwsSdkV2HttpClientStats
+{
+    private final TimeStat connectionAcquireLatency = new TimeStat(MILLISECONDS);
+    private final DistributionStat availableConcurrency = new DistributionStat();
```

Comment:
I'm not sure these make sense as a `DistributionStat` since the connection pool stats of `availableConcurrency`, `leasedConcurrency`, and `pendingConcurrency` are point in time snapshot values and not values that are accumulated. In `AwsSdkClientCoreStats` these are recorded as `AtomicLong` values instead.